src/billing-core/resource-limits.ts

```diff
@@ -0,0 +1,58 @@
+import { BillingProfile } from 'src/billing-core/models';
+
+export interface BillingProfileResourceLimits {
+  availableMachineTypes?: string[];
+  maxAutopause?: number;
+  maxPersistentDiskSize?: number;
+}
+
+/**
+ * Parses a comma-separated string into an array of strings.
+ * If the input is not a string or an empty string, returns undefined.
+ */
+const getCommaSeparatedValues = (value: unknown): string[] | undefined => {
+  if (typeof value === 'string' && value.length !== 0) {
+    return value.split(',');
+  }
+  return undefined;
+};
+
+/**
+ * Parses a value into a number.
+ */
+const getNumberValue = (value: unknown): number | undefined => {
+  if (typeof value === 'number') {
+    return value;
+  }
+  if (typeof value === 'string') {
+    const n = parseInt(value, 10);
+    return Number.isNaN(n) ? undefined : n;
+  }
+  return undefined;
+};
+
+/**
+ * Parse a billing profile's resource limits (if any).
+ *
+ * @param billingProfile - The billing profile.
+ * @returns The billing profile's resource limits, or undefined if the profile does not have any resource limits.
+ */
+export const getResourceLimits = (billingProfile: BillingProfile): BillingProfileResourceLimits | undefined => {
+  if (billingProfile.cloudPlatform !== 'AZURE' || !billingProfile.organization?.limits) {
+    return undefined;
```

Comment:
Might be worth adding a unittest case for GCP.

Reply:
We could probably remove this check for cloud platform entirely. I think this might be a holdover from earlier when this was using billing _projects_ instead of billing _profiles_.

Reply:
Agreed. If these values happen to be set on a profile, so be it.

Reply:
Done in https://github.com/DataBiosphere/terra-ui/pull/4904/commits/490b7b6682ba76db40f1e40143e2f8c3f28011b2.